lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1287,6 +1386,51 @@ class BuildForGroups : public calyx::FuncOpPartialLoweringPattern {
   }
 };
 
+class BuildIfGroups : public calyx::FuncOpPartialLoweringPattern {
+  using FuncOpPartialLoweringPattern::FuncOpPartialLoweringPattern;
+
+  LogicalResult
+  partiallyLowerFuncToComp(FuncOp funcOp,
+                           PatternRewriter &rewriter) const override {
+    LogicalResult res = success();
+    funcOp.walk([&](Operation *op) {
+      if (!isa<scf::IfOp>(op))
+        return WalkResult::advance();
+
+      auto scfIfOp = cast<scf::IfOp>(op);
+
+      calyx::ComponentOp componentOp =
+          getState<ComponentLoweringState>().getComponentOp();
+
+      std::string thenGroupName =
+          getState<ComponentLoweringState>().getUniqueName("then_br");
+      auto thenGroupOp = calyx::createGroup<calyx::GroupOp>(
+          rewriter, componentOp, scfIfOp.getLoc(), thenGroupName);
+      getState<ComponentLoweringState>().setThenGroup(scfIfOp, thenGroupOp);
+
+      if (!scfIfOp.getElseRegion().empty()) {
+        std::string elseGroupName =
+            getState<ComponentLoweringState>().getUniqueName("else_br");
+        auto elseGroupOp = calyx::createGroup<calyx::GroupOp>(
+            rewriter, componentOp, scfIfOp.getLoc(), elseGroupName);
+        getState<ComponentLoweringState>().setElseGroup(scfIfOp, elseGroupOp);
+      }
+
+      for (auto res : scfIfOp.getResults()) {
```

Comment:
Nit: This is confusing since you already have a `res` variable above. Can you rename this please?